packages/dev/core/src/Collisions/gpuPicker.ts

```diff
@@ -147,11 +178,15 @@ export class GPUPicker {
      * Execute a picking operation
      * @param x defines the X coordinates where to run the pick
      * @param y defines the Y coordinates where to run the pick
-     * @param scene defines the scene to pick from
-     * @param disposeWhenDone defines a boolean indicating we do not want to keep resources alive
+     * @param disposeWhenDone defines a boolean indicating we do not want to keep resources alive (false by default)
      * @returns A promise with the picking results
      */
-    public pickAsync(x: number, y: number, scene: Scene, disposeWhenDone = true): Promise<Nullable<AbstractMesh>> {
+    public pickAsync(x: number, y: number, disposeWhenDone = false): Promise<Nullable<AbstractMesh>> {
+        if (!this._pickableMeshes || this._pickableMeshes.length === 0) {

```

Comment:
I wonder if we should not do this instead, to simplify user life:
```typescript
if (!this._pickableMeshes) {
    this.setPickingList(this._cachedScene!.meshes);
}
if (!this._pickableMeshes || this._pickableMeshes.length === 0) {
    return Promise.resolve(null);
}
```
What do you think?

Reply:
Yes so I thought about that but it is an exception and I would prefer to not make it more complex. I find it better to always let the user assume it has to provide the list.